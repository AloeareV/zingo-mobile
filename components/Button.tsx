/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { useTheme } from '@react-navigation/native';

const Button: React.FunctionComponent<any> = ({ type, title, disabled, onPress, style }) => {
  const { colors } = useTheme();
  // type: 'Primary' or 'Secondary'
  const styleButton =
    type === 'Primary'
      ? {
          backgroundColor: disabled ? colors.primaryDisabled : colors.primary,
        }
      : type === 'Secondary'
      ? {
          borderColor: colors.primary || colors.text,
          borderWidth: 2,
        }
      : {
          // error
          backgroundColor: colors.primary,
        };
  const styleCommon = {
    padding: 10,
    paddingLeft: 20,
    paddingRight: 20,
    borderRadius: 10,
    maxWidth: '90%',
    minWidth: '30%',
  };

  return (
    <TouchableOpacity
      style={{
        ...styleButton,
        ...styleCommon,
        ...style,
      }}
      onPress={() => !disabled && onPress && onPress()}>
      <Text
        style={{
          color: type === 'Primary' ? colors.background : colors.primary || colors.text,
          fontWeight: 'bold',
          textTransform: 'uppercase',
          textAlign: 'center',
        }}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
