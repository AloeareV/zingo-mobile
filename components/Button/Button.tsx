/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { useTheme } from '@react-navigation/native';

import { ThemeType } from '../../app/types';

const Button: React.FunctionComponent<any> = ({
  type,
  title,
  disabled,
  onPress,
  style,
  accessible,
  accessibilityLabel,
}) => {
  const { colors } = useTheme() as unknown as ThemeType;
  // type: 'Primary' or 'Secondary'
  const styleButton =
    type === 'Primary'
      ? {
          backgroundColor: disabled ? colors.primaryDisabled : colors.primary,
        }
      : type === 'Secondary'
      ? {
          borderColor: colors.primary || colors.text,
          borderWidth: 2,
        }
      : {
          // error
          backgroundColor: colors.primary,
        };
  const styleCommon = {
    padding: 10,
    paddingLeft: 20,
    paddingRight: 20,
    borderRadius: 10,
    maxWidth: '90%',
    minWidth: '30%',
    minHeight: 48,
  };

  return (
    <TouchableOpacity
      accessible={accessible}
      accessibilityLabel={accessibilityLabel}
      style={{
        ...styleButton,
        ...styleCommon,
        ...style,
      }}
      onPress={() => !disabled && onPress && onPress()}>
      <Text
        style={{
          color: type === 'Primary' ? colors.background : colors.primary || colors.text,
          fontWeight: 'bold',
          textTransform: 'uppercase',
          textAlign: 'center',
        }}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
