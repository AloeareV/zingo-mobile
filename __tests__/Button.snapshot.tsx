/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import Button from '../components/Button';

jest.useFakeTimers()

// test suite
describe("Component Button - test", () => {
  //snapshot test
  test("Button Primary - snapshot", () => {
    const button = create(<Button type={'Primary'} title={'Primary'} disabled={false} onPress={() => {}} style={{}} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  test("Button Secondary - snapshot", () => {
    const button = create(<Button type={'Secondary'} title={'Secondary'} disabled={false} onPress={() => {}} style={{}} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

});
