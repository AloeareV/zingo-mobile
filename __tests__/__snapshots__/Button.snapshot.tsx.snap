// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button - test Button Primary - snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(0, 122, 255)",
      "borderRadius": 10,
      "opacity": 1,
      "padding": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "40%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(242, 242, 242)",
        "fontWeight": "bold",
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Primary
  </Text>
</View>
`;

exports[`Component Button - test Button Secondary - snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "rgb(28, 28, 30)",
      "borderRadius": 10,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "40%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontWeight": "bold",
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Secondary
  </Text>
</View>
`;
